test/sqllogictest/persist-fast-path.slt

```diff
@@ -0,0 +1,98 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+mode cockroach
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET persist_fast_path_limit = 100
+----
+COMPLETE 0
+
+# Verify that the persist fast path only kicks in when it's expected to do so.
+
+statement ok
+CREATE TABLE events (
+    content text,
+    insert_ms numeric,
+    delete_ms numeric
+);
+
+# Applies when the limit is below some threshold. Mapping and
+# projecting is fine.
+
+query T multiline
+EXPLAIN SELECT * from events limit 10;
+----
+Explained Query (fast path):
+  Finish limit=10 output=[#0..=#2]
+    PeekPersist materialize.public.events
+
+EOF
+
+query T multiline
+EXPLAIN SELECT insert_ms + 1000 from events LIMIT 10;
+----
+Explained Query (fast path):
+  Finish limit=10 output=[#0]
+    Project (#3)
+      Map ((#1 + 1000))
+        PeekPersist materialize.public.events
+
+EOF
+
+# Does not apply when the limit is high, or when mixed with features
+# that might require a full scan.
+
+query T multiline
+EXPLAIN SELECT * from events LIMIT 1000;
+----
+Explained Query:
+  Finish limit=1000 output=[#0..=#2]
+    ReadStorage materialize.public.events
+
+EOF
+
+query T multiline
+EXPLAIN SELECT * from events ORDER BY insert_ms LIMIT 10;
+----
+Explained Query:
+  Finish order_by=[#1 asc nulls_last] limit=10 output=[#0..=#2]
+    ReadStorage materialize.public.events
+
+EOF
+
+query T multiline
+EXPLAIN SELECT * from events WHERE delete_ms > mz_now() LIMIT 10;
+----
+Explained Query:
+  Finish limit=10 output=[#0..=#2]
+    Filter (numeric_to_mz_timestamp(#2) > mz_now())
+      ReadStorage materialize.public.events
+
+Source materialize.public.events
+  filter=((numeric_to_mz_timestamp(#2) > mz_now()))
+
+EOF
```

Comment:
Thank you for the tests. A test with `OFFSET` would be nice to have -- the optimization currently does not kick in, even though theoretically it could:

```
materialize=> EXPLAIN SELECT * from events limit 1 offset 2;
               Optimized Plan               
--------------------------------------------
 Explained Query:                          +
   Finish limit=1 offset=2 output=[#0..=#2]+
     ReadStorage materialize.public.events +
 
(1 row)
```

Reply:
Offset was easy enough to support, so I've enabled it and added some tests.